hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/recordcache/ListRecordSource.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state.recordcache;
+
+import static com.hedera.node.app.spi.records.RecordCache.isChild;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.TransactionID;
+import com.hedera.hapi.node.transaction.TransactionReceipt;
+import com.hedera.hapi.node.transaction.TransactionRecord;
+import com.hedera.node.app.spi.records.RecordSource;
+import com.hedera.node.config.data.BlockStreamConfig;
+import com.hedera.node.config.types.StreamMode;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * A {@link RecordSource} that uses a list of precomputed {@link TransactionRecord}s. Used in some tests and when
+ * {@link BlockStreamConfig#streamMode()} is {@link StreamMode#BLOCKS} to  support queryable partial records after
+ * reconnect or restart.
+ */
+public class ListRecordSource implements RecordSource {
```

Comment:
This is basically Records from Receipts right ? Basically `PartialRecordSource`

Reply:
True, renamed.